helm/thingsboard/templates/initializedb-job.yaml

```
@@ -28,11 +28,13 @@ spec:
       imagePullSecrets:
         - name: '{{ .Release.Name }}-container-registry'
       initContainers:
+        {{- if (index .Values "postgresql-ha" "enabled") }}
         - name: check-postgres-db-ready
           image: postgres:{{ index .Values "postgresql-ha" "postgresqlImage" "tag" }}
           command: ['sh', '-c',
             'until pg_isready -h {{ include "thingsboard.pgpoolservicename" . }} -p {{ index .Values "postgresql-ha" "pgpool" "containerPort" }} -t {{ .Values.postgresInitDB.job.check.timeout }};
```

```suggestion
            'until pg_isready -h {{ .Values.postgres.external.hostname  }} -p {{{ .Values.postgres.external.port }} -t {{ .Values.postgresInitDB.job.check.timeout }};
```
 It would be something like this

---

What would be the value for the "image" property?

---

But that image won't be running in this case, right?

---

And anyway, I don't think it's the best way to do this, since the external database is either reachable or not, but it's not like it's going to be unreachable at first and eventually become reachable. There has to be a way to do the check only once (with this we would need it in all jobs that need to wait for the database to be deployed before starting) and fail early with better error messages.

As part of my continuous educations I'm learning K8S. Can we merge this PR and close this ticket, and just make a new PR when I find a more suitable solutions as I progress with my study?

---

afaik there are times where an external db will not be reachable then become reachable (it happened to me in local while tb-node would take a bit to start because the db was not reachable at first)

---

let's do/discuss in another PR

---

Commenting here in case I missed the followup PR. I already asked this same question in the [PR](https://github.com/midokura/evp-chart/pull/211#issuecomment-1398016969) that implemented the pg_check in an init container in the evp api.

Init containers are used to load necessary external resources before the pod starts (e.g. secrets from bitwarden, pulling a file from the cloud, whatever, etc.). I usually ask myself the following questions to understand if something needs to go in an init container or as a runtime check in the health endpoint: can the condition happen once the application has started? if it's no, then the check/operation can go in the init container. if it's a yes, then it needs to go as a health check. 

In this case, can the database become healthy/unhealthy during the life of the application?

I understand that adding this check in the init container was done as an optimization to start a cluster from scratch on a resource constrained environment (CI) that made the startup of pods to fail because of resource starvation. But could we try to find a solution to this problem that is more aligned to k8s best practices? I'd look at startup probes.